pallets/msa/src/lib.rs

```diff
@@ -1129,6 +1134,49 @@ impl<T: Config> Pallet<T> {
 		})
 	}
 
+	/// Modify delegation's schema permissions
+	///
+	/// # Errors
+	/// * [`Error::ExceedsMaxSchemaGrantsPerDelegation`]
+	pub fn upsert_schema_permissions(
+		provider_id: ProviderId,
+		delegator_id: DelegatorId,
+		schema_ids: Vec<SchemaId>,
+	) -> DispatchResult {
+		Self::try_mutate_delegation(delegator_id, provider_id, |delegation, _is_new_delegation| {
```

Comment:
I don't see a mutation of `delegation` happening in here. Am I missing something?